In [143]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
# for dirname, _, filenames in os.walk('/kaggle/input'):
#     for filename in filenames:
#         print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [144]:
import matplotlib.pyplot as plt
import seaborn as sns


In [145]:
df_club=pd.read_csv('../input/mind-graph/Clubs_data.csv')
df_club.head()

Club_Name            Name           Event         Role
0    club_3       Darien Ma  club_3_event_2  Participant
1    club_2    Sege Schrock  club_2_event_1  Participant
2    club_3      Alison bci  club_3_event_2  Participant
3    club_1  kiara jauregui  club_1_event_1  Participant
4    club_2       Tete Buim  club_2_event_3  Participant

In [146]:
df_club.isnull().sum()

Club_Name    0
Name         0
Event        0
Role         0
dtype: int64

In [147]:
df_club.shape

(1046, 4)

In [148]:
df_club['Name'].nunique()

987

In [149]:
df_meta=pd.read_csv('../input/mind-graph/Metadata.csv')
df_meta.head()

Name          ID
0    Annette Ahmed  18XJ1A0100
1   Zachery Clancy  18XJ1A0101
2   Jaquelin Duran  18XJ1A0102
3  Orlando Clemens  18XJ1A0103
4   Dorothy Mayhew  18XJ1A0104

In [150]:
df_meta.isnull().sum()

Name    0
ID      0
dtype: int64

In [151]:
print(df_meta['Name'].nunique())
df_meta.shape

500


(500, 2)

In [152]:
df_org=pd.read_csv('../input/mind-graph/Organisers_In_Fests.csv')
df_org.head()

Fest_Name                Name         Role
0    fest_2       Stcerr Hbmond  organiser_2
1    fest_2         Ayliwn Slks  organiser_6
2    fest_2  Sebestien Hickjman  organiser_7
3    fest_2     Grechen Chlewsn  organiser_8
4    fest_1            Evelin B  organiser_1

In [153]:
print(df_org['Name'].nunique())
df_org.shape

20


(20, 3)

In [154]:
df_org['Name']

0          Stcerr Hbmond
1            Ayliwn Slks
2     Sebestien Hickjman
3        Grechen Chlewsn
4               Evelin B
5      leniez Gilbertson
6       mRiver Flipcieno
7               Leona Ca
8              Ay Kehnei
9          tbdulelh Hors
10             S Stephon
11         Prxry Pefryen
12             Derii Dai
13                    RB
14            Jordi Taem
15                    AM
16      Reshawn Balsdeas
17      Bline Mckesnezie
18        Ashanti John K
19        zoie mcdermott
Name: Name, dtype: object

In [155]:
df_org.isnull().sum()

Fest_Name    0
Name         0
Role         0
dtype: int64

In [156]:
df_part=pd.read_csv('../input/mind-graph/Participants_In_Fests.csv')
df_part.head()

Fest_Name            Name            Event
0    fest_1      JenapRowli   fest_1_event_9
1    fest_1   Credyj Duenas   fest_1_event_4
2    fest_2  Jequqlin Duran   fest_2_event_1
3    fest_1  Keiyah widalgo   fest_1_event_4
4    fest_2   Elaine John S  fest_2_event_15

In [157]:
print(df_part['Name'].nunique())
df_part.shape

4060


(4769, 3)

In [158]:
df_part.isnull().sum()

Fest_Name    0
Name         0
Event        0
dtype: int64

In [159]:
#Import
import fuzzywuzzy
from fuzzywuzzy import fuzz
from fuzzywuzzy import process

In [160]:
df1_names = list(df_meta['Name'].unique())
df2_names = list(df_org['Name'].unique())

In [161]:
def match_names(name, list_names, min_score=0):
    max_score = -1
    max_name = ''
    for x in list_names:
        score = fuzz.partial_ratio(name, x)
        if (score > min_score) & (score > max_score):
            max_name = x
            max_score = score
    return (max_name, max_score)

In [162]:
names = {}
for x in df2_names:
    match = match_names(x, df1_names, 45)
    if match[1] >=45:
        name =  str(match[0]) 
        names[x]=name

names

{'Stcerr Hbmond': 'Starr Hammond',
 'Ayliwn Slks': 'Aylin Seals',
 'Sebestien Hickjman': 'Sebastien Hickman',
 'Grechen Chlewsn': 'Gretchen Clawson',
 'Evelin B': 'Evelin Bradley',
 'leniez Gilbertson': 'Delanie Gilbertson',
 'mRiver Flipcieno': 'River Feliciano',
 'Leona Ca': 'Leona Cabral',
 'Ay Kehnei': 'Aya Kennedy',
 'tbdulelh Hors': 'Abdullah Horst',
 'S Stephon': 'Stephon Smoot',
 'Prxry Pefryen': 'Perry Perryman',
 'Derii Dai': 'Daria Daily',
 'RB': 'Genevieve Ratliff',
 'Jordi Taem': 'Jordi Tam',
 'AM': 'Annette Ahmed',
 'Reshawn Balsdeas': 'Rashawn Balderas',
 'Bline Mckesnezie': 'Blaine Mckenzie',
 'Ashanti John K': 'Ashanti Koenig',
 'zoie mcdermott': 'Zoie McDermott'}

In [163]:
df_org.Name = df_org.Name.map(names)

In [164]:
df_org['Name']

0          Starr Hammond
1            Aylin Seals
2      Sebastien Hickman
3       Gretchen Clawson
4         Evelin Bradley
5     Delanie Gilbertson
6        River Feliciano
7           Leona Cabral
8            Aya Kennedy
9         Abdullah Horst
10         Stephon Smoot
11        Perry Perryman
12           Daria Daily
13     Genevieve Ratliff
14             Jordi Tam
15         Annette Ahmed
16      Rashawn Balderas
17       Blaine Mckenzie
18        Ashanti Koenig
19        Zoie McDermott
Name: Name, dtype: object

In [165]:
df_org['Name'].nunique()

20

repeating the above process for df_part and df_club

In [166]:
df3_names = list(df_part['Name'].unique())
names = []
for x in df3_names:
    match = match_names(x, df1_names, 45)
    if match[1] >= 45:
        name = ((  str(x), str(match[0])  ))
        names.append(name)
name_part_dict = dict(names)


In [167]:
name_part_dict

{'JenapRowli': 'Jayna Rowley',
 'Credyj Duenas': 'Candy Duenas',
 'Jequqlin Duran': 'Jaquelin Duran',
 'Keiyah widalgo': 'Kaliyah Hidalgo',
 'Elaine John S': 'Elaine Sell',
 'Koljte Roend': 'Kolten Roland',
 'Sebestn sHickman': 'Sebastien Hickman',
 'GC': 'Zachery Clancy',
 'JevonteTrevibno': 'Javonte Trevino',
 'Luca S': 'Luca Sanders',
 'Iebelm Perals': 'Isabell Perales',
 'Mcknzie Frn': 'Mckenzie Finn',
 'HlyStroum': 'Holly Strom',
 'Monric Grabowhsy': 'Monica Grabowski',
 'Yesminez Sdler': 'Yasmine Sadler',
 'eyleigh Ktoenig': 'Kayleigh Koenig',
 'Shawn Ka': 'Shawn Kay',
 'Luiz McLxeuglin': 'Luiz McLaughlin',
 'Thee Mcoloud': 'Thea McCloud',
 'Elaine S': 'Elaine Sell',
 'vRlehul Va': 'Rahul Vera',
 'Judikt Cok': 'Judith Crook',
 'Yisrojal Stor': 'Yisroel Storm',
 'gerald arreola': 'Gerald Arreola',
 'Shawn John K': 'Shawn Kay',
 'uoedyBeder': 'Kody Bader',
 'CeylaRicnon': 'Cayla Rincon',
 'Gissase Thorntn': 'Gisselle Thornton',
 'Sereniy Cyestilo': 'Serenity Castillo',
 'DoiencMedi

In [168]:
df_part['Name'] = df_part['Name'].replace(name_part_dict)

In [169]:
len(name_part_dict)

3903

In [170]:
df_part['Name'].nunique()

542

In [171]:
df_part['Name'].isnull().sum()

0

In [172]:
df_part['Name']

0          Jayna Rowley
1          Candy Duenas
2        Jaquelin Duran
3       Kaliyah Hidalgo
4           Elaine Sell
             ...       
4764    Candace Barkley
4765    Debra Clevenger
4766      Theron Stroud
4767     Branson Harman
4768      Jadon Vandyke
Name: Name, Length: 4769, dtype: object

In [173]:
df4_names = list(df_club['Name'].unique())
names = []
for x in df4_names:
    match = match_names(x, df1_names, 45)
    if match[1] >= 45:
        name = ((  str(x), str(match[0])  ))
        names.append(name)
name_club_dict = dict(names)


In [174]:
name_club_dict

{'Darien Ma': 'Darien Martens',
 'Sege Schrock': 'Sage Schrock',
 'Alison bci': 'Alison Mcvey',
 'kiara jauregui': 'Kiara Jauregui',
 'Tete Buim': 'Tate Blum',
 'Loufis emuls': 'Louis Samuels',
 'MercezloWeber': 'Marcello Weber',
 'hJeine uMcCormack': 'Janine McCormack',
 'Quinton Ta': 'Quinton Taylor',
 'Jjnine McCormack': 'Janine McCormack',
 'Everardo John C': 'Everardo Cronin',
 'Were Scholfield': 'Warren Schofield',
 'Ezpre Lod': 'Ezra Lord',
 'Aya K': 'Aya Kennedy',
 'Alannah C': 'Alannah Chisholm',
 'Gremysan eim': 'Grayson Heim',
 'NG': 'Adrian Graff',
 'Judithee Howend': 'Judith Howland',
 'Milo Grafeild': 'Milo Greenfield',
 'A Elena': 'Abigayle Bolen',
 'HeylieGajri': 'Haylie Garay',
 'Sen Fairbaenks': 'Sana Fairbanks',
 'oEvelin rerdli': 'Evelin Bradley',
 'denise minor': 'Denise Minor',
 'Sentagok Nadeau': 'Santiago Nadeau',
 'Grady S': 'Grady Sample',
 'CxhhelseyDeen': 'Chelsey Dean',
 'JM': 'Jaquelin Duran',
 'eriuws Albers': 'Darrius Albers',
 'Annabelle Ma': 'Annabelle

In [175]:
len(name_club_dict)

943

In [176]:
df_club['Name'].nunique()

987

In [177]:
df_club['Name'] = df_club['Name'].replace(name_club_dict)

In [178]:
df_club['Name'].nunique()

304

In [179]:
df_club['Role'].value_counts()

Participant    1001
organiser_1       9
organiser_5       9
organiser_3       9
organiser_4       9
organiser_2       9
Name: Role, dtype: int64

In [180]:
df_club['Name'].isnull().sum()

0

In [181]:
df_club['Name']

0        Darien Martens
1          Sage Schrock
2          Alison Mcvey
3        Kiara Jauregui
4             Tate Blum
             ...       
1041        Kalen Logan
1042     Kathrine Leone
1043        Ibrahim Sun
1044    Milo Greenfield
1045        Daisha Yoon
Name: Name, Length: 1046, dtype: object

In [182]:
df_orgmeta=pd.merge(df_meta,df_org,how='left',on='Name')
df_orgmeta

Name          ID Fest_Name         Role
0      Annette Ahmed  18XJ1A0100    fest_2  organiser_4
1     Zachery Clancy  18XJ1A0101       NaN          NaN
2     Jaquelin Duran  18XJ1A0102       NaN          NaN
3    Orlando Clemens  18XJ1A0103       NaN          NaN
4     Dorothy Mayhew  18XJ1A0104       NaN          NaN
..               ...         ...       ...          ...
495     Andrew Keefe  17XJ1A0345       NaN          NaN
496    Valeria Quinn  17XJ1A0346       NaN          NaN
497   Joselyn Barrow  17XJ1A0347       NaN          NaN
498     Yisroel Sisk  17XJ1A0348       NaN          NaN
499     Natalia Huff  17XJ1A0349       NaN          NaN

[500 rows x 4 columns]

In [183]:
df_orgmeta.isnull().sum()

Name           0
ID             0
Fest_Name    480
Role         480
dtype: int64

In [184]:
df_replacement=df_orgmeta

In [185]:
df_orgmeta.dtypes

Name         object
ID           object
Fest_Name    object
Role         object
dtype: object

In [186]:
df_orgmeta=df_orgmeta.dropna()

In [187]:
df_orgmeta

Name          ID Fest_Name          Role
0         Annette Ahmed  18XJ1A0100    fest_2   organiser_4
7           Aya Kennedy  18XJ1A0107    fest_2   organiser_1
14    Genevieve Ratliff  18XJ1A0114    fest_1   organiser_5
34        Starr Hammond  18XJ1A0134    fest_2   organiser_2
88       Evelin Bradley  18XJ1A0188    fest_1   organiser_1
224     River Feliciano  18XJ1A0324    fest_1   organiser_2
231         Daria Daily  18XJ1A0331    fest_1   organiser_3
235           Jordi Tam  18XJ1A0335    fest_1   organiser_4
265       Stephon Smoot  17XJ1A0115    fest_2   organiser_3
276    Rashawn Balderas  17XJ1A0126    fest_2   organiser_5
282      Abdullah Horst  17XJ1A0132    fest_1   organiser_6
299         Aylin Seals  17XJ1A0149    fest_2   organiser_6
305   Sebastien Hickman  17XJ1A0155    fest_2   organiser_7
317     Blaine Mckenzie  17XJ1A0167    fest_1   organiser_7
345    Gretchen Clawson  17XJ1A0195    fest_2   organiser_8
363      Ashanti Koenig  17XJ1A0213    fest_1   organiser_8
377      Zoie McDermott  17XJ1A0227    fest_2   organiser_9
380  Delanie Gilbertson  17XJ1A0230    fest_2  organiser_10
406        Leona Cabral  17XJ1A0256    fest_1   organiser_9
467      Perry Perryman  17XJ1A0317    fest_1  organiser_10

In [188]:
df_part.head()

Fest_Name             Name            Event
0    fest_1     Jayna Rowley   fest_1_event_9
1    fest_1     Candy Duenas   fest_1_event_4
2    fest_2   Jaquelin Duran   fest_2_event_1
3    fest_1  Kaliyah Hidalgo   fest_1_event_4
4    fest_2      Elaine Sell  fest_2_event_15

In [189]:
df_part['Fest_Name'].value_counts()

fest_2    2745
fest_1    2024
Name: Fest_Name, dtype: int64

In [190]:
df_metaclub=pd.merge(df_meta,df_club,how='left',on='Name')

In [191]:
df_metaclub

Name          ID Club_Name           Event         Role
0     Annette Ahmed  18XJ1A0100    club_2  club_2_event_2  Participant
1     Annette Ahmed  18XJ1A0100    club_1  club_1_event_1  organiser_1
2     Annette Ahmed  18XJ1A0100    club_1  club_1_event_2  Participant
3     Annette Ahmed  18XJ1A0100    club_1  club_1_event_2  organiser_1
4     Annette Ahmed  18XJ1A0100    club_1  club_1_event_3  Participant
...             ...         ...       ...             ...          ...
1230   Yisroel Sisk  17XJ1A0348    club_1  club_1_event_1  Participant
1231   Yisroel Sisk  17XJ1A0348    club_2  club_2_event_1  Participant
1232   Yisroel Sisk  17XJ1A0348    club_2  club_2_event_2  Participant
1233   Yisroel Sisk  17XJ1A0348    club_3  club_3_event_1  Participant
1234   Natalia Huff  17XJ1A0349       NaN             NaN          NaN

[1235 rows x 5 columns]

In [192]:
df_metaclub.isnull().sum()

Name           0
ID             0
Club_Name    240
Event        240
Role         240
dtype: int64

In [193]:
df_metapart=pd.merge(df_meta,df_part,how='left',on='Name')
df_metapart['Role']='Participant'
df_metapart

Name          ID Fest_Name            Event         Role
0     Annette Ahmed  18XJ1A0100    fest_2  fest_2_event_10  Participant
1     Annette Ahmed  18XJ1A0100    fest_2   fest_2_event_3  Participant
2     Annette Ahmed  18XJ1A0100    fest_2   fest_2_event_8  Participant
3     Annette Ahmed  18XJ1A0100    fest_2   fest_2_event_4  Participant
4     Annette Ahmed  18XJ1A0100    fest_2  fest_2_event_13  Participant
...             ...         ...       ...              ...          ...
4645   Natalia Huff  17XJ1A0349    fest_2   fest_2_event_1  Participant
4646   Natalia Huff  17XJ1A0349    fest_2  fest_2_event_12  Participant
4647   Natalia Huff  17XJ1A0349    fest_1   fest_1_event_5  Participant
4648   Natalia Huff  17XJ1A0349    fest_2   fest_2_event_7  Participant
4649   Natalia Huff  17XJ1A0349    fest_2   fest_2_event_2  Participant

[4650 rows x 5 columns]

In [194]:
df_metapart.isnull().sum()

Name           0
ID             0
Fest_Name    115
Event        115
Role           0
dtype: int64

In [195]:
df_orgmeta

Name          ID Fest_Name          Role
0         Annette Ahmed  18XJ1A0100    fest_2   organiser_4
7           Aya Kennedy  18XJ1A0107    fest_2   organiser_1
14    Genevieve Ratliff  18XJ1A0114    fest_1   organiser_5
34        Starr Hammond  18XJ1A0134    fest_2   organiser_2
88       Evelin Bradley  18XJ1A0188    fest_1   organiser_1
224     River Feliciano  18XJ1A0324    fest_1   organiser_2
231         Daria Daily  18XJ1A0331    fest_1   organiser_3
235           Jordi Tam  18XJ1A0335    fest_1   organiser_4
265       Stephon Smoot  17XJ1A0115    fest_2   organiser_3
276    Rashawn Balderas  17XJ1A0126    fest_2   organiser_5
282      Abdullah Horst  17XJ1A0132    fest_1   organiser_6
299         Aylin Seals  17XJ1A0149    fest_2   organiser_6
305   Sebastien Hickman  17XJ1A0155    fest_2   organiser_7
317     Blaine Mckenzie  17XJ1A0167    fest_1   organiser_7
345    Gretchen Clawson  17XJ1A0195    fest_2   organiser_8
363      Ashanti Koenig  17XJ1A0213    fest_1   organiser_8
377      Zoie McDermott  17XJ1A0227    fest_2   organiser_9
380  Delanie Gilbertson  17XJ1A0230    fest_2  organiser_10
406        Leona Cabral  17XJ1A0256    fest_1   organiser_9
467      Perry Perryman  17XJ1A0317    fest_1  organiser_10

In [196]:
df_onlyfest=pd.merge(df_orgmeta,df_metapart,how='outer',on=['Name','Role','ID','Fest_Name'])
df_onlyfest

Name          ID Fest_Name         Role            Event
0         Annette Ahmed  18XJ1A0100    fest_2  organiser_4              NaN
1           Aya Kennedy  18XJ1A0107    fest_2  organiser_1              NaN
2     Genevieve Ratliff  18XJ1A0114    fest_1  organiser_5              NaN
3         Starr Hammond  18XJ1A0134    fest_2  organiser_2              NaN
4        Evelin Bradley  18XJ1A0188    fest_1  organiser_1              NaN
...                 ...         ...       ...          ...              ...
4665       Natalia Huff  17XJ1A0349    fest_1  Participant  fest_1_event_13
4666       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_4
4667       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_3
4668       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_8
4669       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_5

[4670 rows x 5 columns]

In [197]:
df_onlyfest.isnull().sum()

Name           0
ID             0
Fest_Name    115
Role           0
Event        135
dtype: int64

In [198]:
df_onlyfest['participated']=True
for i in range(0,20):
    df_onlyfest['participated'][i]=False

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [199]:
df_onlyfest

Name          ID Fest_Name         Role            Event  \
0         Annette Ahmed  18XJ1A0100    fest_2  organiser_4              NaN   
1           Aya Kennedy  18XJ1A0107    fest_2  organiser_1              NaN   
2     Genevieve Ratliff  18XJ1A0114    fest_1  organiser_5              NaN   
3         Starr Hammond  18XJ1A0134    fest_2  organiser_2              NaN   
4        Evelin Bradley  18XJ1A0188    fest_1  organiser_1              NaN   
...                 ...         ...       ...          ...              ...   
4665       Natalia Huff  17XJ1A0349    fest_1  Participant  fest_1_event_13   
4666       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_4   
4667       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_3   
4668       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_8   
4669       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_5   

      participated  
0            False  
1            False  
2            False  
3            False  
4            False  
...            ...  
4665          True  
4666          True  
4667          True  
4668          True  
4669          True  

[4670 rows x 6 columns]

dropping all persons who are not in any fest

marking events for first 20 none

In [200]:
for i in range(0,20):
    df_onlyfest['Event'][i]='none'

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [201]:
# df_onlyfest['Event']=df_onlyfest['Event'].astype(str)
# for index,rows in df_onlyfest.iterrows():
#     if len(rows['Event']==0):
#         df_onlytest.drop(index,inplace=True)

In [202]:
df_newfest=df_onlyfest.dropna()
df_newfest

Name          ID Fest_Name         Role            Event  \
0         Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
1           Aya Kennedy  18XJ1A0107    fest_2  organiser_1             none   
2     Genevieve Ratliff  18XJ1A0114    fest_1  organiser_5             none   
3         Starr Hammond  18XJ1A0134    fest_2  organiser_2             none   
4        Evelin Bradley  18XJ1A0188    fest_1  organiser_1             none   
...                 ...         ...       ...          ...              ...   
4665       Natalia Huff  17XJ1A0349    fest_1  Participant  fest_1_event_13   
4666       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_4   
4667       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_3   
4668       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_8   
4669       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_5   

      participated  
0            False  
1            False  
2            False  
3            False  
4            False  
...            ...  
4665          True  
4666          True  
4667          True  
4668          True  
4669          True  

[4555 rows x 6 columns]

In [203]:
# df_onlyfest=df_onlyfest.fillna('none')

In [204]:
df_onlyfest

Name          ID Fest_Name         Role            Event  \
0         Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
1           Aya Kennedy  18XJ1A0107    fest_2  organiser_1             none   
2     Genevieve Ratliff  18XJ1A0114    fest_1  organiser_5             none   
3         Starr Hammond  18XJ1A0134    fest_2  organiser_2             none   
4        Evelin Bradley  18XJ1A0188    fest_1  organiser_1             none   
...                 ...         ...       ...          ...              ...   
4665       Natalia Huff  17XJ1A0349    fest_1  Participant  fest_1_event_13   
4666       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_4   
4667       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_3   
4668       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_8   
4669       Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_5   

      participated  
0            False  
1            False  
2            False  
3            False  
4            False  
...            ...  
4665          True  
4666          True  
4667          True  
4668          True  
4669          True  

[4670 rows x 6 columns]

In [216]:
df_metaclub['Name'].value_counts()

Annette Ahmed        16
Jaquelin Duran       14
Irvin Lira           10
Litzy Moyer           9
Javier Breaux         9
                     ..
Charlotte Lorenzo     1
Devyn Ewing           1
Thea McCloud          1
Serenity Ngo          1
Nestor Light          1
Name: Name, Length: 260, dtype: int64

In [206]:
df_metaclub.isnull().sum()

Name           0
ID             0
Club_Name    240
Event        240
Role         240
dtype: int64

In [207]:
# df_metaclub['Role']=df_metaclub['Role'].astype(str)
# for index,row in df_metaclub.iterrows():
#     if len(row['Role'])==0:
#         df_metaclub.drop(index,inplace=True,axis=0)
# df_metaclub.isnull().sum()
df_metaclub=df_metaclub.dropna()

In [217]:
df_metaclub

Name          ID Club_Name           Event         Role
0      Annette Ahmed  18XJ1A0100    club_2  club_2_event_2  Participant
1      Annette Ahmed  18XJ1A0100    club_1  club_1_event_1  organiser_1
2      Annette Ahmed  18XJ1A0100    club_1  club_1_event_2  Participant
3      Annette Ahmed  18XJ1A0100    club_1  club_1_event_2  organiser_1
4      Annette Ahmed  18XJ1A0100    club_1  club_1_event_3  Participant
...              ...         ...       ...             ...          ...
1229  Joselyn Barrow  17XJ1A0347    club_1  club_1_event_3  Participant
1230    Yisroel Sisk  17XJ1A0348    club_1  club_1_event_1  Participant
1231    Yisroel Sisk  17XJ1A0348    club_2  club_2_event_1  Participant
1232    Yisroel Sisk  17XJ1A0348    club_2  club_2_event_2  Participant
1233    Yisroel Sisk  17XJ1A0348    club_3  club_3_event_1  Participant

[995 rows x 5 columns]

In [212]:
club_1=0
club_2=0
club_3=0
for index,row in df_metaclub.iterrows():
    if row['Club_Name']=='club_1' and row['Role']=='Participant':
        club_1=club_1+1
    elif row['Club_Name']=='club_2' and row['Role']=='Participant':
        club_2=club_2+1
    elif row['Club_Name']=='club_3' and row['Role']=='Participant':
        club_3=club_3+1
print(club_1)
print(club_2)
print(club_3)

317
319
317


In [215]:
df_metaclub['Role'].value_counts()

Participant    953
organiser_1      9
organiser_3      9
organiser_5      8
organiser_2      8
organiser_4      8
Name: Role, dtype: int64

In [214]:
fest_1=0
fest_2=0

for index,row in df_onlyfest.iterrows():
    if row['Fest_Name']=='fest_1' and row['Role']=='Participant':
        fest_1=fest_1+1
    elif row['Fest_Name']=='fest_2' and row['Role']=='Participant':
        fest_2=fest_2+1
#     elif row['Club_Name']=='club_3' and row['Role']=='Participant':
#         club_3=club_3+1
print(fest_1)
print(fest_2)
# print(club_3)

1912
2623


clearly we can see that there are more people perticipating in fests rather than club activities
953 participants and 42 organisers

In [209]:
df_onlyfest['Role'].value_counts()

Participant     4650
organiser_4        2
organiser_1        2
organiser_5        2
organiser_2        2
organiser_3        2
organiser_6        2
organiser_7        2
organiser_8        2
organiser_9        2
organiser_10       2
Name: Role, dtype: int64

In [218]:
df_onlyfest['Name'].value_counts()

Keyara Camacho     54
Jaquelin Duran     50
Zachery Clancy     49
Annette Ahmed      41
Branson Harman     40
                   ..
Ciera Almanza       1
Lewis Breedlove     1
Katelynn Harlow     1
Shea Jenkins        1
Kobe Barrientos     1
Name: Name, Length: 500, dtype: int64

4650 participants in fests and 20 organisers for total events

In [210]:
df_onlyfest['Event'].value_counts()

fest_1_event_3     210
fest_2_event_4     210
fest_2_event_2     210
fest_1_event_4     208
fest_2_event_5     208
fest_1_event_5     208
fest_2_event_1     208
fest_1_event_2     207
fest_1_event_1     207
fest_2_event_3     201
fest_2_event_10    142
fest_2_event_14    141
fest_2_event_7     140
fest_1_event_7     140
fest_2_event_9     139
fest_1_event_6     139
fest_2_event_11    139
fest_2_event_8     137
fest_1_event_9     136
fest_2_event_12    136
fest_2_event_13    136
fest_1_event_10    135
fest_2_event_15    135
fest_1_event_13    135
fest_2_event_6     134
fest_1_event_8     133
fest_1_event_12    131
fest_1_event_11    130
none                20
Name: Event, dtype: int64

**we can see that even after having less no.of participants clubs are  having more no.of organizers than the no.of organizers in fests, even the participation is less in club events compared to fests**

**defining club heads depending on participation in events**

In [225]:
df_common=pd.merge(df_onlyfest,df_metaclub,how='left',on=['Name','ID'])
df_common

Name          ID Fest_Name       Role_x          Event_x  \
0     Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
1     Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
2     Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
3     Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
4     Annette Ahmed  18XJ1A0100    fest_2  organiser_4             none   
...             ...         ...       ...          ...              ...   
9163   Natalia Huff  17XJ1A0349    fest_1  Participant  fest_1_event_13   
9164   Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_4   
9165   Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_3   
9166   Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_8   
9167   Natalia Huff  17XJ1A0349    fest_1  Participant   fest_1_event_5   

      participated Club_Name         Event_y       Role_y  
0            False    club_2  club_2_event_2  Participant  
1            False    club_1  club_1_event_1  organiser_1  
2            False    club_1  club_1_event_2  Participant  
3            False    club_1  club_1_event_2  organiser_1  
4            False    club_1  club_1_event_3  Participant  
...            ...       ...             ...          ...  
9163          True       NaN             NaN          NaN  
9164          True       NaN             NaN          NaN  
9165          True       NaN             NaN          NaN  
9166          True       NaN             NaN          NaN  
9167          True       NaN             NaN          NaN  

[9168 rows x 9 columns]

In [226]:
df_common.dropna()

Name          ID Fest_Name       Role_x         Event_x  \
0     Annette Ahmed  18XJ1A0100    fest_2  organiser_4            none   
1     Annette Ahmed  18XJ1A0100    fest_2  organiser_4            none   
2     Annette Ahmed  18XJ1A0100    fest_2  organiser_4            none   
3     Annette Ahmed  18XJ1A0100    fest_2  organiser_4            none   
4     Annette Ahmed  18XJ1A0100    fest_2  organiser_4            none   
...             ...         ...       ...          ...             ...   
9148   Yisroel Sisk  17XJ1A0348    fest_2  Participant  fest_2_event_4   
9149   Yisroel Sisk  17XJ1A0348    fest_2  Participant  fest_1_event_2   
9150   Yisroel Sisk  17XJ1A0348    fest_2  Participant  fest_1_event_2   
9151   Yisroel Sisk  17XJ1A0348    fest_2  Participant  fest_1_event_2   
9152   Yisroel Sisk  17XJ1A0348    fest_2  Participant  fest_1_event_2   

      participated Club_Name         Event_y       Role_y  
0            False    club_2  club_2_event_2  Participant  
1            False    club_1  club_1_event_1  organiser_1  
2            False    club_1  club_1_event_2  Participant  
3            False    club_1  club_1_event_2  organiser_1  
4            False    club_1  club_1_event_3  Participant  
...            ...       ...             ...          ...  
9148          True    club_3  club_3_event_1  Participant  
9149          True    club_1  club_1_event_1  Participant  
9150          True    club_2  club_2_event_1  Participant  
9151          True    club_2  club_2_event_2  Participant  
9152          True    club_3  club_3_event_1  Participant  

[5644 rows x 9 columns]

In [227]:
df_common['Name'].value_counts()

Jaquelin Duran       700
Annette Ahmed        656
Zachery Clancy       392
Keyara Camacho       270
Irvin Lira           170
                    ... 
Tori Seibert           1
Maximilian McGraw      1
Annalee Simmons        1
Cameron Burleson       1
Trista Pritchard       1
Name: Name, Length: 500, dtype: int64